src/components/MenuItemCard.test.tsx

```diff
@@ -0,0 +1,141 @@
+import { Menu, MenuList } from '@chakra-ui/react';
+import { fireEvent, screen } from '@testing-library/react';
+
+import { MenuItemCard } from './MenuItemCard';
+import { render } from '../utils/test-utils';
+
+describe('MenuItemCard', () => {
+  describe('when the text color is default', () => {
+    const mockOnClick = jest.fn();
+
+    beforeEach(() => {
+      render(
+        <Menu defaultIsOpen={true}>
+          <MenuList>
+            <MenuItemCard
+              icon={<span>Icon</span>}
+              label="Label"
+              textColor="default"
+              onClick={mockOnClick}
+            />
+          </MenuList>
+        </Menu>,
+      );
+    });
+
+    it('renders the icon and label', () => {
+      const icon = screen.getByText('Icon');
+      const label = screen.getByText('Label');
+      expect(icon).toBeInTheDocument();
+      expect(label).toBeInTheDocument();
+    });
+
+    it('calls the onClick function when clicked', () => {
+      const menuItem = screen.getByRole('menuitem');
+      fireEvent.click(menuItem);
+
+      expect(mockOnClick).toHaveBeenCalled();
+    });
+  });
+
+  describe('when the textColor is red', () => {
+    beforeEach(() => {
+      render(
+        <Menu defaultIsOpen={true}>
+          <MenuList>
+            <MenuItemCard
+              icon={<span>Icon</span>}
+              label="Label"
+              textColor="red"
+            />
+          </MenuList>
+        </Menu>,
+      );
+    });
+
+    it('renders the icon and label', () => {
+      const icon = screen.getByText('Icon');
+      const label = screen.getByText('Label');
+      expect(icon).toBeInTheDocument();
+      expect(label).toBeInTheDocument();
+    });
+  });
+
+  describe('when the onClick function is not provided', () => {
+    beforeEach(() => {
+      render(
+        <Menu defaultIsOpen={true}>
+          <MenuList>
+            <MenuItemCard icon={<span>Icon</span>} label="Label" />
+          </MenuList>
+        </Menu>,
+      );
+    });
+
+    it('does not call the onClick function when clicked', () => {
+      const menuItem = screen.getByRole('menuitem');
+      expect(menuItem).not.toHaveAttribute('onClick');
+    });
+  });
+
+  it('match the snapshot', () => {
+    const { asFragment } = render(
+      <Menu defaultIsOpen={true}>
+        <MenuList>
+          <MenuItemCard icon={<span>Icon</span>} label="Label" />
+        </MenuList>
+      </Menu>,
+    );
+    expect(asFragment()).toMatchInlineSnapshot(`
```

Comment:
The snapshot content was auto generated from `yarn test`. we can't use external snapshot files due to lint rule forced to use `InlineSnapshot`, which make our file and difficult to read.

Reply:
Is there a reason for the snapshot testing?

Reply:
snapshot tests is valuable to verify DOM structure, and can provide visual of outcome of DOM especially in large system. i dont know the unit tests guideline of metamask team, since Metamask Mobile project require snapshot tests for component apart from normal content tests, behaviour tests,  that is why i implement snapshot tests here as well for shared components. .